WordPress/WordPressTest/PrivacySettingsAnalyticsTrackerTests.swift

```diff
@@ -0,0 +1,25 @@
+import Nimble
+@testable import WordPress
+import XCTest
+
+class PrivacySettingsAnalyticsTrackerTests: XCTestCase {
+
+    override func setUp() {
+        super.setUp()
+
+        // Because AnalyticsEventTrackingSpy logs events in a static var, we need to reset it between tests
+        AnalyticsEventTrackingSpy.reset()
+    }
+
+    func test_trackPrivacySettingsReportCrashesToggled_enabled() {
+        let tracker = PrivacySettingsAnalyticsTracker(tracker: AnalyticsEventTrackingSpy.self)
+
+        tracker.trackPrivacySettingsReportCrashesToggled(enabled: true)
+
+        expect(AnalyticsEventTrackingSpy.trackedEvents).to(haveCount(1))
+        expect(AnalyticsEventTrackingSpy.trackedEvents).to(containElementSatisfying({ event in
+            event.name == PrivacySettingsAnalytics.privacyChoicesBannerSaveButtonTapped.rawValue &&
+            event.properties["enabled"] == true.stringLiteral
+        }))
+    }
+}
```

Comment:
Testing event logging is awkward and cumbersome. As this PR shows, to do it right one needs to test both the type that tracks the events and the code that uses that tracker.

I don't have a clean alternative in mind at the moment, but I think we could simplify the setup by ditching domain-specific loggers in favor using `WPAnalytics` directly and moving the event configuration logic elsewhere, maybe in a builder. Not sure. As I said, I don't have a clean proposal in mind yet.

At the same time, adding the extra layer of abstraction might be useful if/when we'll move towards self-contained micro apps. E.g. the settings screens could be their own micro app and at that point it might be useful to not access `WPAnalytics` directly.

Reply:
>but I think we could simplify the setup by ditching domain-specific loggers in favor using WPAnalytics directly and moving the event configuration logic elsewhere, maybe in a builder. Not sure. As I said, I don't have a clean proposal in mind yet.

I think this is okay, and in case we need domain-specific logging methods, we can add them to a domain-specific `WPAnalytics` extension. For example:

1. Let's say we need to have a `trackPrivacySettingsReportCrashesToggled` method. 
2. We can add this method to a `WPAnalytics+PrivacySettings.swift` extension instead of adding everything to `WPAnalytics.m` directly.

Same for the events, I think it's great to keep having domain-specific enums ( like `PrivacySettingsAnalytics` enum ) instead of adding all events to a massive enum.